### PR TITLE
feat:added filters context on product listing page

### PR DESCRIPTION
**Pull request by JatuFaique**
image

Added useFilter Context for product listing page

### CODE REVIEW

File: src/components/Filters.js [hunk 1]
```diff
@@ -57,25 +81,93 @@ function Filters() {
...
             <label>
-              <input type="radio" name="sort" />
-              Low to high
+              <input
+                type="radio"
+                name="sort"
+                onChange={(e) => {
+                  filterDispatch({
+                    type: "HIGH_TO_LOW",
+                    payload: "HIGH_TO_LOW",
```
Comment (amirnathani9) [inline src/components/Filters.js:103]: i think here payload is not needed. you can directly acces we action .type

File: src/helper/index.js [hunk 2]
```diff
@@ -19,4 +22,14 @@ const sortByDiscount = (prevProd, minDiscount) => {
   return prevProd.filter((product) => product.discount >= minDiscount);
 };
 
-export { sortByCategory, sortByPrice, sortByDiscount };
+function sortByHL(prevProd, sortBy) {
+  if (sortBy && sortBy === "HIGH_TO_LOW") {
```
Comment (amirnathani9) [inline src/helper/index.js:26]: cant understand what is HL, can give a elaborated names

File: src/helper/index.js [hunk 3]
```diff
@@ -19,4 +22,14 @@ const sortByDiscount = (prevProd, minDiscount) => {
...
+    return [...prevProd.sort((a, b) => b.price - a.price)];
+  }
+  if (sortBy && sortBy === "LOW_TO_HIGH") {
+    return [...prevProd.sort((a, b) => a.price - b.price)];
+  }
+  return prevProd;
+}
+
+export { sortByCategory, sortByPrice, sortByDiscount, sortByHL };
```
Comment (amirnathani9) [inline src/helper/index.js:35]: u can also make a different file for differnt helpers function to look the code cleaner

File: src/pages/ProductsPage.js [hunk 4]
```diff
@@ -22,12 +27,20 @@ function ProductsPage() {
...
+  // console.log("after cat", sortedByCategory);
+  const sortedByDiscount = sortByDiscount(
+    sortedByCategory,
+    filterState.minDiscount
+  ); // 10 , 20 , 30 , 40 ,50 , 60 , 70 , 80
+  // console.log("after dis", sortedByDiscount);
+  const sortedbyHL = sortByHL(sortedByDiscount, filterState.sortBy);
+
+  const finalProd = sortedbyHL;
   // setProductData(sortedByDiscount);
```
Comment (amirnathani9) [inline src/pages/ProductsPage.js:44]: you can do destructuring here for filterState

**Review by amirnathani9 — APPROVED** [review]
The logic is good and the code is clean.
I have commented some things u can make the changes.
I cant see the live Preview to check whether its working or not, plzz provide live preview link.
also one suggestion is write proper commit messages after every changes

File: src/components/Filters.js [hunk 5]
```diff
@@ -57,25 +81,93 @@ function Filters() {
               <span>Sort By</span>
             </div>
             <label>
-              <input type="radio" name="sort" />
+              <input
+                type="radio"
+                name="sort"
+                onChange={(e) => {
+                  filterDispatch({
+                    type: "LOW_TO_HIGH",
```
Comment (nyyshaaa) [inline src/components/Filters.js:89]: you can give the type name as sort by to indicate that it's for sorting as you have done for categories to indicate categories also both type and payload are named same, so can change the name.

File: src/reducers/filterReducer.js [hunk 6]
```diff
@@ -6,8 +8,39 @@ export const initialFilter = {
...
+      }
+
+    case "DISCOUNT":
+      return {
+        ...state,
+        minDiscount: action.payload,
+      };
+    case "HIGH_TO_LOW":
+      console.log("htl");
+      return {
```
Comment (nyyshaaa) [inline src/reducers/filterReducer.js:34]: remove console in case you forgot

**Review by nyyshaaa — APPROVED** [review]
code looks good to me ,you can merge. suggested minor changes .

**Pull request merged**